resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsSnapshot.scala

```diff
@@ -59,6 +60,14 @@ object ExecutorPodsSnapshot extends Logging {
         case "pending" =>
           PodPending(pod)
         case "running" =>
+          // A pod will be considered running as long as there is at least one running container,
+          // so we need to check if there are any failed containers.
+          if (pod.getSpec.getRestartPolicy.equals("Never") &&
+            pod.getStatus.getContainerStatuses.stream
+            .map[ContainerStateTerminated](cs => cs.getState.getTerminated)
+            .anyMatch(t => t != null && t.getExitCode != 0)) {
+              PodFailed(pod)
+          }
           PodRunning(pod)
```

Comment:
Hi, @khogeland 
This PR ignores line 69 and returns `PodRunning(pod)` always. Did I understand correctly?

Reply:
Hm, why do you say line 69 is ignored? The unit test should be proof that this works.

Reply:
So there is no return on PodFailed it will just get computed and then thrown away and replaced with pod running. Does the unit test fail without this change?

Reply:
Oops, you're right, missing `else` here